### ISSUE TITLE
switch style enforcement: additional ideas

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1443]
I often faced developer who write this

```go
switch a {
// expected values
case 1:
case 2:
   // do nothing on purpose
default:
   return errors.New("blah")
}
```

It occurs with people that comes from a language where switch requires a break, otherwise they fallthrought.

I dislike it, because it should be written like this

```go
switch a {
// expected values
case 1, 2:
   // do nothing on purpose
default:
   return errors.New()
}
```

Why ?

Because the code could evolve like this later

```go
switch a {
// expected values
case 1:
case 2:
   logSomething()
default:
   return errors.New()
}
```

logSomething() is only called for 2 ?

Is it on purpose? Or not ?

A decent code review will catch it, but you will have to know the expected use case.

Here, the code change could either be expected to be

```go
switch a {
// expected values
case 2:
   logSomething()
case 1:
   // do nothing on purpose
default:
   return errors.New()
}
```

Or 

```go
switch a {
// expected values
case 1, 2:
   logSomething()
default:
   return errors.New()
}
```

For me, switch case without content should be put together, either at the end or beginning.

If they are multiple one, they should be grouped.

So this

```go
switch a {
case 1:
case 2:
     return 42
case 3:
default:
    return -1
case 4:
case 5:
    return 123
}

return doSomething(a)
```

should be this

```go
switch a {
case 2:
     return 42
case 5:
     return 123
case 1, 3, 4:
default:
    return -1
}

return doSomething(a)
```

Also an empty case should be commented I think

```go
switch a {
case 2:
     return 42
case 1, 3, 4:
     // these are expected values
default:
    return -1
}

return doSomething(a)
```

Another thing could be to allow this, where empty cases are at the end after the default case

```go
switch a {
case 2:
     return 42
case 5:
     return 123
default:
    return -1
case 1, 3, 4:
}

return doSomething(a)
```

What do you guys think about all this?

Is there something to detect this in revive ? Or another linter out of revive?

Unfortunately, I don't have a computer with me, maybe it exists and we can close, ... or it's a new linter idea.

**Comment by ccoVeille** [comment]
Please note that identical-switch-branches only address a part of the problem.

**Comment by chavacava** [comment]
We could start by enforcing comments on empty cases.
`identical-switch-branches` could also evolve to consider comments when comparing branches.
